python-api/app/integrations/google_calendar/fetch_calendar_api_data.py

```
@@ -0,0 +1,189 @@
+        str: The full path to the specified file within the 'data' directory.
+
+    Raises:
+        Exception: If the directory creation fails, an exception is raised
+        with a message indicating the failure.
+    """
+    try:
+        cwd = os.path.dirname(os.path.abspath(__file__)) #  get current working directory
+        data_dir = os.path.join(cwd, "data")
+        os.makedirs(data_dir, exist_ok=True)
+    except OSError as e:
+        raise Exception(f"Failed to create directory {data_dir}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)